client/src/Login.tsx

```diff
@@ -49,10 +49,17 @@ const initialFormData: UserLoginData = {
     email: '',
     password: '',
 };
+
+interface error {
+    type: '' | 'email' | 'password',
+    message: string
+}
+
 function Login() {
     const classes = useStyles();
     const history = useHistory();
-
+    const [isLoading, setIsLoading] = React.useState<boolean>(false)
+    const [error, setError] = React.useState<error>({type: '', message: ''})
```

Comment:
this feels like it should be `error | null`
and `string | null` in the child components that consume, pass down like `error?.message ?? null` or something

`string | false` could be ok but i've just never seen it so there's probs some smell there

Reply:
ah ok, i see we have to check the type first. well see if you can get something like ☝️ to work but no probs if not

Reply:
that may have exposed an anti-pattern. we may want the form controls in the here, but let's figure that out later